File: backend/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace PeerGrid.Backend.Models
{
    public class Transaction
    {
        [Key]
        public int Id { get; set; }

        public int LearnerId { get; set; }
        public int TutorId { get; set; }
        public string Skill { get; set; } = string.Empty;
        public decimal Points { get; set; }
        public DateTime Timestamp { get; set; }

        public double? Rating { get; set; } // 1-5 stars
    }
}

File: backend(Spring Boot)/src/main/java/com/peergrid/backend/entity/Transaction.java
package com.peergrid.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "transactions")
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "learner_id")
    private User learner;

    @ManyToOne
    @JoinColumn(name = "tutor_id")
    private User tutor;

    private String skill = "";
    private BigDecimal points;
    private String type = "Transfer"; // Earned, Spent, Transfer
    private LocalDateTime timestamp;

    private Double rating; // 1-5 stars
}
